public/app/features/admin/ServerStats.tsx

```diff
@@ -1,79 +1,166 @@
-import React, { PureComponent } from 'react';
+import React, { useEffect, useState } from 'react';
 import { hot } from 'react-hot-loader';
 import { connect } from 'react-redux';
-import { Icon, Tooltip } from '@grafana/ui';
-import { NavModel } from '@grafana/data';
-import { StoreState } from 'app/types';
+import { css } from '@emotion/css';
+import { CardContainer, LinkButton, useStyles2 } from '@grafana/ui';
+import { GrafanaTheme2 } from '@grafana/data';
+import { AccessControlAction, StoreState } from 'app/types';
 import { getNavModel } from 'app/core/selectors/navModel';
-import Page from 'app/core/components/Page/Page';
 import { getServerStats, ServerStat } from './state/apis';
+import { contextSrv } from '../../core/services/context_srv';
 
-interface Props {
-  navModel: NavModel;
-  getServerStats: () => Promise<ServerStat[]>;
+export interface Props {
+  getServerStats: () => Promise<ServerStat | null>;
```

Comment:
Any reason for providing this fn as a property? Especially that it's mapped with redux connect, what feels unnecesary. Think our primary pattern for such scenarios is to import the api functions in the component of interest, rather than providing those as a property.

Not sure this is the best examples, but take a look: 
https://github.com/grafana/grafana/blob/3b515e650c55f0125020660d8aa22811f756b0d3/public/app/features/admin/AdminEditOrgPage.tsx#L44
https://github.com/grafana/grafana/blob/10badea19e47572b8667fa65d3985842fa0aef44/public/app/features/dashboard/components/SaveDashboard/SaveDashboardAsModal.tsx#L14

Yet, providing the fn as a property makes the component easier to test as imports don't need to be mocked. Maybe a good candidate for a discussion during Frontend Weekly?

Reply:
This was the old code so I left it as it is, but I think the advantage here is ease of testing, although it's not that much more difficult to mock the api method in the test, so I'll remove this prop.

Reply:
That's totally fine, didn't notice it's a "legacy" :)